services/indexes/params/collections.go

```diff
@@ -64,7 +64,11 @@ type StatisticsParams struct {
 }
 
 func (p *StatisticsParams) ForValues(v uint8, q url.Values) error {
-	return p.ListParams.ForValues(v, q)
+	err := p.ListParams.ForValues(v, q)
+	if p.ListParams.Limit == 5000 {
```

Comment:
As discussed, please add some comments for clarity explaining why the limit is changed here and what does 0 mean.